protocol/dataunit/client_server_test.go

```diff
@@ -153,28 +199,31 @@ func echoServer(t *testing.T, ctx context.Context, s *Server) error {
 			reqCtx, cancel := context.WithCancel(ctx)
 			defer cancel()
 
-			req, r, err := s.ServeDataUnit(reqCtx)
+			data, r, err := s.ServeDataUnit(reqCtx)
 			if err != nil {
-				if err == errTestDone {
+				if errors.Is(err, errTestDone) {
 					return
 				}
-				t.Errorf("echoServer: ServeDataUnit(): err == %v", err)
+				echoServerErr <- fmt.Errorf("echoServer: ServeDataUnit(): err == %w", err)
+				return
 			}
 			time.Sleep(randDuration(10 * time.Millisecond))
-			err = r.RespondDataUnit(reqCtx, req)
+			err = r.RespondDataUnit(reqCtx, data)
 			if err != nil {
-				if err == errTestDone {
+				if errors.Is(err, errTestDone) {
 					return
 				}
-				t.Errorf("echoServer: WriteDataUnit(): err == %v", err)
+				echoServerErr <- fmt.Errorf("echoServer: RespondDataUnit(): err == %w", err)
 			}
 		}()
 	}
 }
 
 func randDuration(max time.Duration) time.Duration {
-	return time.Duration(rand.Int63n(int64(max)))
+	return time.Duration(rand.Int64N(int64(max)))
```

Comment:
Use `rand.N` and avoid all the type conversions. https://pkg.go.dev/math/rand/v2#example-N

Reply:
Cool, will do that 👍🏻